cfme/utils/virtual_machines.py

```diff
@@ -56,10 +56,13 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
 
     if template_name is None:
         try:
-            deploy_args.update(template=provider_crud.data['small_template'])
+            deploy_args.update(template=provider_crud.data['templates']['small_template']['name'])
         except KeyError:
-            raise ValueError('small_template not defined for Provider {} in cfme_data.yaml'.format(
-                provider_key))
+            try:
+                deploy_args.update(template=provider_crud.data['small_template'])
+            except KeyError:
+                raise ValueError('small_template not defined for Provider {} in cfme_data.yaml'
+                    .format(provider_key))
```

Comment:
**REQUIRED**
It's better to fix yaml than the code.

Reply:
I think we can drop it, and fix the KeyError either in yaml or in the specific test.

Reply:
@jan-zmeskal Please drop this 2nd try, and just raise the ValueError directly.